### DISCUSSION TITLE
Have multiple sketches for one mesh

### DISCUSSION THREAD

**Discussion by theoryshaw** [hlorus/CAD_Sketcher#325] [Feature Requests]
**Motivation / Existing Problem**
This is perhaps more relative to [BlenderBIM](https://blenderbim.org/), but I have a feeling this functionality could be useful with a more agnostic use, as well. 

With BlenderBIM's implementation of IFC, a mesh can have multiple profiles associated with.  Example [here.](https://github.com/hlorus/CAD_Sketcher/issues/26#issuecomment-1109745905)

With the [current 'alpha' integration](https://github.com/hlorus/CAD_Sketcher/issues/26) of BlenderBIM & CAD_Sketcher, one mesh can only have one CS sketch associated with it. 

**Description**
This feature request proposes developing an interface where you're able to modify multiple profiles per mesh object.

**Mockup**
The following example has (5) profiles associated with one mesh object.  The extrusion depth of the profile could come from [create_extruded_area_solid](https://github.com/IfcOpenShell/IfcOpenShell/blob/5fb574cf5c1a208da36e1d3a2589bd82b37d7d59/src/blenderbim/blenderbim/bim/module/geometry/helper.py#L329-L339) function.

![2022-10-06_10-58-45_Blender_blender](https://user-images.githubusercontent.com/507113/194361741-8c6657d0-89e9-42b3-9b99-fc875a93342a.gif)

**Comment by hlorus** [comment]
I don't really get your proposal here, with boolean modifiers it is already possible to create a mesh from multiple sketches, doesn't that fit your needs?

**Comment by Moult** [comment]
In IFC, a single object may be made out of multiple extrusions, each with its own profile. I'm not entirely sure how / if / what CAD Sketcher can do to make this type of integration more flexible, but let's perhaps keep this bug on hold until a workflow is more fleshed out. Potentially there is nothing to be done?

**Comment by hlorus** [comment]
Sounds reasonable 👍🏼
